Cabal/src/Distribution/Simple/Build/PathsModule/Z.hs

```diff
@@ -101,17 +118,62 @@ render z_root = execWriter $ do
     tell "catchIO = Exception.catch\n"
     return ()
   tell "\n"
+  tell "-- |The package version.\n"
   tell "version :: Version\n"
   tell "version = Version "
   tell (zVersionDigits z_root)
   tell " []\n"
   tell "\n"
+  tell "-- |If the argument is a filename listed in the @data-files@ field of the\n"
```

Comment:
What if the filename *isn't* listed in the @data-files@ field?

Looking at the code, it looks like this function just unilaterally returns a filepath, whether or not the file exists.

Reply:
@chreekat, I followed (from https://cabal.readthedocs.io/en/stable/cabal-package-description-file.html#accessing-data-files-from-package-code):
> `getDataFileName :: FilePath -> IO FilePath`
> 
> If the argument is a filename listed in the `data-files` field, the result is the name of the corresponding file on the system on which the program is running.

but I am happy to update both.

Reply:
I think updating both seems good since it's a bit confusing.

Reply:
Now done.